operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

```diff
@@ -127,56 +123,66 @@ private void addPodSpecTemplate(JobSpecFluent<?> builder, Keycloak keycloak, Con
     }
 
     private PodSpec createPodSpec(Context<Keycloak> context) {
-        var allVolumes = getAllVolumes(context);
-        Collection<String> requiredVolumes = new HashSet<>();
-        var builder = new PodSpecBuilder();
+        StatefulSet current = ContextUtils.getCurrentStatefulSet(context).orElseThrow();
+        StatefulSet desired = ContextUtils.getDesiredStatefulSet(context);
+
+        // start off with the desired statefulset state
+        var builder = desired.getSpec().getTemplate().getSpec().edit();
         builder.withRestartPolicy("Never");
-        addInitContainer(builder, context, allVolumes.keySet(), requiredVolumes);
-        addContainer(builder, context, allVolumes.keySet(), requiredVolumes);
+
+        // remove things we don't want
+
+        // TODO: this may actually not be correct - via the unsupported PodTemplate someone could make configuration dependent upon an init container
+        builder.withInitContainers();
```

Comment:
Copying over init containers might be a good thing. But as PodTemplate is not supported, might do a best effort here. 

Reading up on init containers, it seems that they run sequentially. So we could then add ours at the end.